src/Extensions/SiteTreeSubsites.php

```diff
@@ -400,11 +400,10 @@ public static function contentcontrollerInit($controller)
 
         $ignore_subsite_locale = Config::inst()->get(self::class, 'ignore_subsite_locale');
 
-        if ($subsite && $subsite->Language) {
-            if (!$ignore_subsite_locale && $subsite && i18n::getData()->validate($subsite->Language)) {
-                i18n::set_locale($subsite->Language);
-            }
+        if (!$ignore_subsite_locale && $subsite && i18n::getData()->validate($subsite->Language) && $subsite->Language) {
```

Comment:
```suggestion
        if (!$ignore_subsite_locale && $subsite && $subsite->Language && i18n::getData()->validate($subsite->Language)) {
```
If there's no language we shouldn't pass the non-existent language into the validate method.